clang/lib/Tooling/DependencyScanning/DependencyScanningWorker.cpp

```diff
@@ -449,7 +449,11 @@ class DependencyScanningAction : public tooling::ToolAction {
       Opts->Targets = {
           deduceDepTarget(ScanInstance.getFrontendOpts().OutputFile,
                           ScanInstance.getFrontendOpts().Inputs)};
-    Opts->IncludeSystemHeaders = true;
+    if (Format == ScanningOutputFormat::Make) {
+      // Only 'Make' scanning needs to force this because that mode depends on
+      // getting the dependencies directly from \p DependencyFileGenerator.
+      Opts->IncludeSystemHeaders = true;
```

Comment:
Can this change go upstream?

Reply:
My intention was to do [this change](https://github.com/apple/llvm-project/blob/f0fc1cfca77dd7a03fd51bcb4e20351130d84376/clang/lib/Tooling/DependencyScanning/DependencyScanningWorker.cpp#L460) upstream, which will then correct this in a better way.
I'm a bit reluctant to directly make this PR's change upstream because it is not testable there.

Reply:
Oh nice, sounds like a better solution.